contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,275 @@
+  Inodes *uint64   
+}  
+
+// PodUsage holds pod-level metrics.  
+type PodUsage struct {  
+  // UID of the pod  
+  PodUID string  
+  // Metrics pertaining to pod total usage of cpu  
+  // This may include additional overhead not included in container usage statistics.  
+  CPU *CpuUsage  
+  // Metrics pertaining to pod total usage of system memory  
+  // This may include additional overhead not included in container usage statistics.  
+  Memory *MemoryUsage   
+  // Metrics of containers in the pod.  
+  Containers []ContainerUsage  
+  // Metrics pertaining to volume usage of filesystem resources.  
+  Volumes []VolumeUsage  
+}  
+
+// ContainerUsage holds container-level usage metrics.  
+type ContainerUsage struct {  
+  // ID of the container  
+  ContainerID string  
+  // Metrics pertaining to container usage of cpu  
+  CPU *CpuUsage  
+  // Metrics pertaining to container usage of system memory  
+  Memory *MemoryUsage  
+  // Metrics pertaining to container rootfs usage of disk.  
+  // Rootfs.UsedBytes is the number of bytes used for the container write layer.  
+  Rootfs *FilesystemUsage   
+  // Metrics pertaining to container logs usage of Disk.  
+  Logs *FilesystemUsage  
+}  
+
+// CpuUsage holds statistics about the amount of cpu time consumed  
+type CpuUsage struct {  
+  // The time at which these Metrics were updated.  
+  Timestamp metav1.Time  
+  // Average CPU usage rate over sample window (across all cores), in "nano cores".  
+  // The "core" unit represents nanoseconds of CPU time consumed per second.  
+  // For example, 5 nanocores means the process averaged 5 nanoseconds 
+  // of cpu time per second during the sample window.
+  UsageRateNanoCores *uint64  
+  // Cumulative CPU usage (sum of all cores) since object creation.  
+  AggregateUsageCoreNanoSeconds *uint64   
+}  
+
+// MemoryUsage holds statistics about the quantity of memory consumed  
+type MemoryUsage struct {  
+  // The time at which these metrics were updated.  
+  Timestamp metav1.Time  
+  // The amount of working set memory. This includes recently accessed memory,  
+  // dirty memory, and kernel memory.  
+  WorkingSetBytes *uint64  
+}  
+
+// VolumeUsage holds statistics about the quantity of disk resources consumed for a volume  
+type VolumeUsage struct {  
+  // Embedded FilesystemUsage  
+  FilesystemUsage  
+  // Name is the name given to the Volume  
+  Name string  
+}  
+
+// FilesystemUsage holds statistics about the quantity of disk resources consumed  
+type FilesystemUsage struct {  
+  // The time at which these metrics were updated.  
+  Timestamp metav1.Time  
+  // The name filesystme on which resources are consumed.  This must uniquely identify the filesystem.
+  Filesystem string  
+  // UsedBytes represents the disk space consumed on the filesystem, in bytes.  
+  UsedBytes *uint64  
+  // InodesUsed represents the inodes consumed on the filesystem  
+  InodesUsed *uint64  
```

Comment:
Bikeshedding: the order inconsistency between `UsedBytes` and `InodesUsed` seems a bit off. `UsedInodes` instead?